Packs/AWS-SecurityHub/Integrations/AWSSecurityHubEventCollector/AWSSecurityHubEventCollector.py

```diff
@@ -0,0 +1,205 @@
+    )
+
+
+def main():
+    params = demisto.params()
+    args = demisto.args()
+    command = demisto.command()
+
+    aws_role_arn = params.get('role_arn')
+    aws_role_session_name = params.get('role_session_name')
+    aws_default_region = params.get('default_region')
+    aws_role_session_duration = params.get('session_duration')
+    aws_access_key_id = params.get('credentials', {}).get('identifier')
+    aws_secret_access_key = params.get('credentials', {}).get('password')
+    verify_certificate = not params.get('insecure', True)
+    timeout = params.get('timeout')
+    retries = params.get('retries', 5)
+
+    # How much time before the first fetch to retrieve events
+    first_fetch_time: datetime.datetime = arg_to_datetime(
+        arg=params.get('first_fetch', '3 days'),
+        arg_name='First fetch time',
+        required=True
+    )
+
+    try:
+        validate_params(
+            aws_default_region=aws_default_region,
+            aws_role_arn=aws_role_arn,
+            aws_role_session_name=aws_role_session_name,
+            aws_access_key_id=aws_access_key_id,
+            aws_secret_access_key=aws_secret_access_key,
+        )
+
+        aws_client = AWSClient(
+            aws_default_region=aws_default_region,
+            aws_role_arn=aws_role_arn,
+            aws_role_session_name=aws_role_session_name,
+            aws_role_session_duration=aws_role_session_duration,
+            aws_role_policy=None,
+            aws_access_key_id=aws_access_key_id,
+            aws_secret_access_key=aws_secret_access_key,
+            verify_certificate=verify_certificate,
+            timeout=timeout,
+            retries=retries,
+        )
+
+        client = aws_client.aws_session(
+            service='securityhub',
+            region=params.get('region'),
+            role_arn=params.get('roleArn'),
+            role_session_name=params.get('roleSessionName'),
+            role_session_duration=params.get('roleSessionDuration'),
+        )
+
+        if command == 'test-module':
+            # This is the call made when pressing the integration Test button.
+            get_events(client, limit=1)
+            return_results('ok')
+
+        elif command == 'aws-securityhub-get-events':
+            should_push_events = argToBoolean(args.get('should_push_events'))
+            return_results(get_events_command(client=client, should_push_events=should_push_events))
+
+        elif command == 'fetch-events':
+            next_run, events = fetch_events(
+                client=client,
+                last_run=demisto.getLastRun(),
+                first_fetch_time=first_fetch_time
+            )
+
+            # Saves next_run for the time fetch-events is invoked
+            demisto.setLastRun(next_run)
+
+            send_events_to_xsiam(
+                events,
+                vendor=VENDOR,
+                product=PRODUCT
+            )
```

Comment:
I think it's better to move it to another function, e.g. `fetch_events_command`
so the main function will be smaller